sklearn/base.py

```diff
@@ -323,6 +323,53 @@ def _get_tags(self):
                 collected_tags.update(more_tags)
         return collected_tags
 
+    def _check_warm_start_with(self, warm_start_with):
+        # Return True if should warm start
+
+        if not hasattr(self, '_warmstartable_parameters'):
+            raise ValueError(
+                "None of the {} parameters can be warm-started."
+                .format(self.__class__.__name__)
+            )
+
+        if not warm_start_with:  # None or empty dict
+            return False
+
+        def find_param(param_name):
+            # util to make sure the param is warm-startable, and to know
+            # whether the param must be increase or decrease while
+            # warm-started.
+            for warmstartable_param in self._warmstartable_parameters:
+                if param_name == warmstartable_param[1:]:
+                    must_increase = (warmstartable_param[0] == '+')
+                    return must_increase
+
+            # No match found, raise error
+            raise ValueError(
+                "The {} parameter cannot be warm-started."
+                .format(param_name)
+            )
+
+        err_msg = (
+            "The {} class can only be warm-started with {} "
+            "values of {}. Current value is {}, requesting new value "
+            "of {}."
+        )
+        for param_name, new_value in warm_start_with.items():
+            current_value = self.get_params()[param_name]
+            must_increase = find_param(param_name)
+            direction = 'increasing' if must_increase else 'decreasing'
+            if ((must_increase and new_value < current_value) or
```

Comment:
not sure how to handle a warm start on the exact previous param values.  Maybe we want it to be `<=` instead of `<`?

Reply:
In master, we allow for `<=`.

Pretty sure that raising an error on `<=` would make it impossible to start with warm-start=True

Reply:
Ah yeah, sorry, I mistakenly thought we only accept the new parameter if it's strictly larger or smaller than the previous one. This looks good.